### ISSUE TITLE
GridRaySampler should not persist the grid.

### ISSUE THREAD

**Issue by PeterL1n** [facebookresearch/pytorch3d#802]
## 🚀 Feature

`GridRaySampler` creates the grid in constructor and calls `register_buffer`. However, we should pass `persistent=False` argument so that the canonical grid is not saved in state_dict. 

https://github.com/facebookresearch/pytorch3d/blob/804117833e3c02c19a6774f70bac1a780a322228/pytorch3d/renderer/implicit/raysampling.py#L99

## Motivation

We want the grid to be generated in code every time instead of loading from the state_dict. This allows us to change the image render size at the construction time. Currently, the state_dict will mismatch if the image size is changed. This is a needed feature since it is common to have training and evaluation to be at different resolutions.

## Pitch

Change
```py
self.register_buffer("_xy_grid", _xy_grid)
```
to
```py
self.register_buffer("_xy_grid", _xy_grid, False)
```

But note that this may break existing state_dict's as the new module no longer has `_xy_grid`, but previously saved state_dict will have the extra key. Which can be workaround by using `load_state_dict(state_dict, strict=False)` What are your thoughts on BC breaking?

**Comment by bottler** [comment]
I agree that this would be a good enhancement. We're thinking about the BC-breaking implications internally.

**Comment by PeterL1n** [comment]
@bottler 

Maybe an approach is to add another arg in the constructor `persist_grid=True` by default to be BC compatible. Then people who need interoperability can set it to `False` to prevent the grid from being persisted...

**Comment by bottler** [comment]
Asking people to set `strict=False` is no more bothersome than asking them to change the code which creates their model. I'm wondering if there is a way to have a custom deserializer/unpickler in our code for the GridRaySampler which explicitly ignores any provided _xy_grid, which I think would be the nicest.

**Comment by PeterL1n** [comment]
@bottler Maybe something like this.


```py
def _load_from_state_dict(self, state_dict, prefix, *args, **kwargs):
      del state_dict[prefix + '_xy_grid']
      super()._load_from_state_dict(state_dict, prefix, *args, **kwargs)
```

**Comment by bottler** [comment]
Yes. That is one of several ways to fix the problem, relying on internal parts of PyTorch. Another is to register a hook with `self._register_load_state_dict_pre_hook`.

We have decided that we prefer to avoid such things and just have a documented BC break in PyTorch3D. I plan to make the change soon.
